test/JDBC/src/main/java/com/sqlsamples/batch_run.java

```diff
@@ -297,6 +296,12 @@ static void batch_run_sql(Connection con_bbl, BufferedWriter bw, String testFile
                         checkParallelQueryExpected = true;
                         continue;
                     }
+
+                    if (strLine.toLowerCase().startsWith("-- db_collation_expected")){
+                        checkDbCollationExpected = true;
+                        continue;
+                    }
```

Comment:
This approach of adding `-- db_collation_expected` has following disadvantages
- Developers will have additional work of adding this flag into files which are having different expected output with non-default db_collation.
- More prone to mistakes, as developers might remove the expected file from db_collation and might miss removing this flag, causing test failures.

Instead we can have approach as follows,
- When db_collation is not "default", JDBC test framework should check for the expected output file of a test in `db_collation` folder, if exists that expected output will be used for diff, else expected output will be picked from `expected` folder.
- In this approach, developers adding test do not have to worry about adding the flag `db_collation_expected` in the test files. We followed this approach in non-default server collation JDBC test framework, you can refer to it.

Reply:
Updated in latest rev